packages/edit-site/src/components/global-styles/hooks.js

```diff
@@ -105,10 +163,10 @@ export function useSetting( path, blockName, source = 'all' ) {
 	let result;
 	switch ( source ) {
 		case 'all':
-			result = get( userConfig, finalPath ) ?? getBaseSetting();
+			result = get( userConfig, finalPath, {} ).user ?? getBaseSetting();
```

Comment:
This change is confusing to me. Why do we have a `.user` subkey in the configs?

Reply:
Also, this change makes it so that `all` and `user` are basically similar and we don't want that: `all` is meant to returned the merged configs, while `user` is meant to return only the user changes (excluding the base config)

Reply:
This seems to be related to having `.user` as values for some configs, now that we can easily access all "user", "base" values directly maybe we can normalize all configs and drop this sunkey entirely.

Reply:
I think we still need the subkey, base includes core and theme so we need the subkey to differentiate e.g: the UI will show both. So I guess we can normalize to always have subkeys as it is doing now.

Reply:
> the UI will show 

The UI can do

```
useSetting( 'something', context, 'all' ); // merged
useSetting( 'something', context, 'user' ); // user
useSetting( 'something', context, 'base' ); // base
``` 

so the object itself shouldn't have the keys. it's just adds complexity and it's not consistent across settings and styles (and even inside settings)

Reply:
How would the UI differentiate between core and theme palettes? The base needs to have subjects for core and theme, the UI shows them differently.

Reply:
It could be a different API call (query arg or something). Anyway, it feels like implementation detail for now but I feel it would simplify a lot of things if the "configs" had a consistent shape, regardless how they're retrieved and the arguments (merged, user, base,...).

Reply:
> it would simplify a lot of things if the "configs" had a consistent shape, regardless of how they're retrieved and the


I agree with that but I think we are being consistent if all the presets no matter where they come from are grouped by origin e.g: in base object we have { core: [], theme: [] }, in user we have { user: [] }, in the merged object we have { core: [], theme: [], user: [] }.

Grouping by origin we have a consistent shape where we can always work on.

> This change is confusing to me. Why do we have a .user subkey in the configs?

The addition of the user subkey is done in order to improve consistency we make presets always have an origin subkey.

But I think I understand the point that in an user object having a user subkey is redundant it is also ok for me if the user object does not have the subkey as long as the base ( { core: [], theme: [] } ) and merged objects ( { core: [], theme: [], user: [] } ) have them.

Reply:
> But I think I understand the point that in an user object having a user subkey is redundant it is also ok for me if the user object

Yes, this is the issue I have with this but it's not specific to the user object for me, it's for all objects. We should have a "core object",  a "theme object", a "user object" and a "merged object" all with no subkeys.

Reply:
> We should have a "core object", a "theme object", a "user object" and a "merged object" all with no subkeys.

On the block editor, we are also going to show different color palettes per origin. So we thought it would be better to pass a single object to the block editor where presets are keyed by origin instead of passing multiple objects and that by origin subkey approach appeared. We can have multiple objects but in that case, we also need to make the block editor aware of the multiple objects, the block editor will never need anything from other sources besides the presets so the by origin approach seemed a better fit.